codecov_auth/views/okta.py

```diff
@@ -179,15 +114,29 @@ def _login_user(self, request: HttpRequest, iss: str, user_data: dict):
         login(request, current_user)
         return current_user
 
-    def get(self, request):
+    def validate_issuer(self) -> str | None:
+        """Checks that the issuer is valid. If not, it returns None."""
+        iss = settings.OKTA_ISS
+        if iss is None:
+            log.warning("Unable to log in due to missing Okta issuer", exc_info=True)
+            return None
+        if not ISS_REGEX.match(iss):
+            log.warning("Invalid Okta issuer")
+            return None
+        return iss
+
+    def get(self, request: HttpRequest) -> HttpResponse:
+        iss = self.validate_issuer()
+        if iss is None:
+            return redirect(f"{settings.CODECOV_DASHBOARD_URL}/login")
```

Comment:
If I read this right, the only variable here is `settings.OKTA_ISS`, which I believe should be an immutable deployment configuration.

Does it make sense to run the validation on every request in that case?
Also, does it make sense to do a redirect to `/login` in case okta is misconfigured? Although whether you display a "okta login is not enabled" error page, or redirect to the login page does not matter that much.

From a SaaS product perspective: is there a hard SLO for the okta integration? In which case it would make sense to assert on deployment that a valid issuer is configured.

---

Long story short, I see you are just moving existing code around, so this is not a regression in itself, just some in-depth questions I’m asking :-)

Reply:
> Does it make sense to run the validation on every request in that case?

Probably not :p Ideally, we would just check this on initialization and use it as it for the rest of the requests. I didn't want to modify this (because I was unsure about the consequences), which is why I just took the conservative approach to leave it for now (especially this isn't really related to my goal).

> Also, does it make sense to do a redirect to /login in case okta is misconfigured?

It's going to be hard to tell from the Okta response whether it's mis-configured. For admins that are integrating with Okta, they will probably need to test it out before deploying it. We probably don't want to leak that the integration is mis-configured to non-admin users though, so redirecting them to the login page is probably fine.

> From a SaaS product perspective: is there a hard SLO for the okta integration?

AFAICT, this Okta integration is _only_ for self-hosted. We don't support this on Codecov's core cloud platform, so I doubt there's any SLAs around this. The Okta integration that I'm adding in Part 2 will be different though, and I think we're potentially going to provide feedback to the users integrating on the admin page if their login integration has worked or not. Not sure if we're going to add an SLA around this, but hopefully there will be more feedback.